database/skylink.go

```diff
@@ -1,11 +1,53 @@
 package database
 
 import (
+	"fmt"
 	"time"
 
+	"gitlab.com/SkynetLabs/skyd/skymodules"
+	"go.mongodb.org/mongo-driver/bson/bsontype"
 	"go.mongodb.org/mongo-driver/bson/primitive"
+	"go.mongodb.org/mongo-driver/x/bsonx/bsoncore"
+	"go.sia.tech/siad/crypto"
 )
 
+// Hash is a struct that embeds the crypto.Hash, allowing us to implement the
+// bsoncodec ValueMarshaler interfaces.
+type Hash struct {
```

Comment:
This looks a bit strange. Didn't `type Hash crypto.Hash` work?

Reply:
resolving as we discussed it in 1-1, had to embed to access methods